### ISSUE TITLE
ContentEncodingTest.testIdentity crash on macosX64

### ISSUE THREAD

**Issue by cy6erGn0m** [ktorio/ktor#1194]
### Ktor Version
1.2.2

### Ktor Engine Used(client or server and name)
IosClientEngine

`Uncaught Kotlin exception: kotlin.RuntimeException: Unexpected receiver type: noname`

```

Uncaught Kotlin exception: kotlin.RuntimeException: Unexpected receiver type: noname
    at 0   test.kexe                           0x000000010e112e87 kfun:kotlin.Throwable.<init>(kotlin.String?)kotlin.Throwable + 23 (/Users/teamcity/buildAgent/work/4d622a065c544371/runtime/src/main/kotlin/kotlin/Throwable.kt:21:56)
    at 1   test.kexe                           0x000000010e112e45 kfun:kotlin.Exception.<init>(kotlin.String?)kotlin.Exception + 21 (/Users/teamcity/buildAgent/work/4d622a065c544371/runtime/src/main/kotlin/kotlin/Exceptions.kt:23:5)
    at 2   test.kexe                           0x000000010e112e15 kfun:kotlin.RuntimeException.<init>(kotlin.String?)kotlin.RuntimeException + 21 (/Users/teamcity/buildAgent/work/4d622a065c544371/runtime/src/main/kotlin/kotlin/Exceptions.kt:34:5)
    at 3   test.kexe                           0x000000010e112d24 kfun:kotlin.native.internal.ThrowInvalidReceiverTypeException(kotlin.reflect.KClass<#STAR>)kotlin.Nothing + 164 (/Users/teamcity/buildAgent/work/4d622a065c544371/runtime/src/main/kotlin/kotlin/native/internal/RuntimeUtils.kt:32:11)
    at 4   test.kexe                           0x000000010e253c68 kfun:io.ktor.client.utils.buildHeaders(kotlin.Function1<io.ktor.http.HeadersBuilder,kotlin.Unit>)io.ktor.http.Headers + 312 (/Users/teamcity/buildAgent/work/4d622a065c544371/backend.native/build/stdlib/kotlin/util/Standard.kt:83:5)
    at 5   test.kexe                           0x000000010e266293 kfun:io.ktor.client.engine.ios.IosClientEngine.<no name provided>_2.URLSession_5#internal + 691 (/Users/user.name/ktor/ktor-client/ktor-client-ios/darwin/src/io/ktor/client/engine/ios/IosClientEngine.kt:56:31)
    at 6   test.kexe                           0x000000010e2674be _knbridge37 + 174 (/Users/user.name/ktor/ktor-client/ktor-client-ios/darwin/src/io/ktor/client/engine/ios/IosClientEngine.kt:<unknown>)
    at 7   CFNetwork                           0x00007fff3bc7041a __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.163 + 255
    at 8   Foundation                          0x00007fff3eded55c __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
    at 9   Foundation                          0x00007fff3eded464 -[NSBlockOperation main] + 68
    at 10  Foundation                          0x00007fff3edc31dd -[__NSOperationInternal _start:] + 685
    at 11  Foundation                          0x00007fff3eded197 __NSOQSchedule_f + 227
    at 12  libdispatch.dylib                   0x00007fff68a38cea _dispatch_block_async_invoke2 + 83
    at 13  libdispatch.dylib                   0x00007fff68a2d63d _dispatch_client_callout + 8
    at 14  libdispatch.dylib                   0x00007fff68a3868d _dispatch_main_queue_callback_4CF + 1135
    at 15  CoreFoundation                      0x00007fff3cb50cd7 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
    at 16  CoreFoundation                      0x00007fff3cb50401 __CFRunLoopRun + 2289
    at 17  CoreFoundation                      0x00007fff3cb4f8be CFRunLoopRunSpecific + 455
    at 18  Foundation                          0x00007fff3edb38ef -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 280
    at 19  Foundation                          0x00007fff3ee83bea -[NSRunLoop(NSRunLoop) runUntilDate:] + 114
    at 20  test.kexe                           0x000000010e1b157a kfun:io.ktor.client.tests.utils.dispatcher.$testSuspend$lambda-1COROUTINE$18.invokeSuspend#internal + 794 (/Users/user.name/ktor/ktor-client/ktor-client-tests/ktor-client-tests-dispatcher/macosX64/src/TestMacosX64.kt:23:9)
    at 21  test.kexe                           0x000000010e137e75 kfun:kotlin.coroutines.native.internal.BaseContinuationImpl.resumeWith(kotlin.Result<kotlin.Any?>) + 325 (/Users/teamcity/buildAgent/work/4d622a065c544371/runtime/src/main/kotlin/kotlin/coroutines/ContinuationImpl.kt:30:39)
    at 22  test.kexe                           0x000000010e13cbce kfun:kotlinx.coroutines.DispatchedTask.run() + 1102 (/Users/teamcity/buildAgent/work/4d622a065c544371/backend.native/build/stdlib/kotlin/util/Result.kt:151:0)
    at 23  test.kexe                           0x000000010e13deb3 kfun:kotlinx.coroutines.EventLoopImpl.processNextEvent()kotlin.Long + 323 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/EventLoop.kt:96:16)
    at 24  test.kexe                           0x000000010e1aeec2 kfun:kotlinx.coroutines.BlockingCoroutine.joinBlocking#internal + 242 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Builders.kt:63:44)
    at 25  test.kexe                           0x000000010e1aea2d kfun:kotlinx.coroutines.runBlocking(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,#GENERIC>)Generic + 893 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Builders.kt:48:22)
    at 26  test.kexe                           0x000000010e1ae660 kfun:kotlinx.coroutines.runBlocking$default(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,#GENERIC>;kotlin.Int)Generic + 176 (/opt/buildAgent/work/44ec6e850d5c63f0/kotlinx-coroutines-core/native/src/Builders.kt:31:8)
    at 27  test.kexe                           0x000000010e1ae514 kfun:io.ktor.client.tests.utils.dispatcher.testSuspend(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,kotlin.Unit>) + 116 (/Users/user.name/ktor/ktor-client/ktor-client-tests/ktor-client-tests-dispatcher/macosX64/src/TestMacosX64.kt:<unknown>)
    at 28  test.kexe                           0x000000010e1ae45f kfun:io.ktor.client.tests.utils.dispatcher.testSuspend$default(kotlin.coroutines.CoroutineContext;kotlin.coroutines.SuspendFunction1<kotlinx.coroutines.CoroutineScope,kotlin.Unit>;kotlin.Int) + 159 (/Users/user.name/ktor/ktor-client/ktor-client-tests/ktor-client-tests-dispatcher/macosX64/src/TestMacosX64.kt:<unknown>)

```

**Comment by cy6erGn0m** [comment]
Should be fixed in K/N JetBrains/kotlin-native#3093

**Comment by e5l** [comment]
Fixed in kotlin-native `master`.
